plugins/database/redshift/redshift.go

```diff
@@ -0,0 +1,528 @@
+	// this username
+	var dbname sql.NullString
+	if err := db.QueryRowContext(ctx, "SELECT current_database();").Scan(&dbname); err != nil {
+		return err
+	}
+
+	if dbname.Valid {
+		/*
+			- SP means we can't have parallel sessions creating and destroying the SP
+			- SP means we may accidentially leave a SP on the database if there's a network partition
+			- Since we've already revoked all the client privs, and those revocations are instant, we should
+			rely on the client killing its own connection instead of us doing it for them
+			- For large redshift clusters, upper bound on connections is 5000
+			- For small redshift clusters, upper bound on connections is less than 1000 (need to double check)
+			- Sessions are memory bound, so this may be an issue if clients fail to exit their session
+			- Scenario: TTL of 1 minute with hundreds of users means we may have hundreds of zombie connections
+			  Reality: TTLs are longer lived since apps can't handle reconnecting this often anyways, so it's probably ok?
+		*/
+		revocationStmts = append(revocationStmts, `CREATE OR REPLACE PROCEDURE terminateloop(dbusername varchar(100))
+LANGUAGE plpgsql
+AS $$
+DECLARE
+  currentpid int;
+  loopvar int;
+  qtyconns int;
+BEGIN
+SELECT COUNT(process) INTO qtyconns FROM stv_sessions WHERE user_name=dbusername;
+  FOR loopvar IN 1..qtyconns LOOP
+    SELECT INTO currentpid process FROM stv_sessions WHERE user_name=dbusername ORDER BY process ASC LIMIT 1;
+    SELECT pg_terminate_backend(currentpid);
+  END LOOP;
+END
+$$;`)
+
+		revocationStmts = append(revocationStmts, fmt.Sprintf(`call terminateloop('%s');`, username))
+		//revocationStmts = append(revocationStmts, `DROP PROCEDURE terminateloop(varchar);`)
+	}
+
+	fmt.Printf("%+v\n", revocationStmts)
+
+	// again, here, we do not stop on error, as we want to remove as
+	// many permissions as possible right now
+	var lastStmtError error
+	for _, query := range revocationStmts {
+		if err := dbtxn.ExecuteDBQuery(ctx, db, nil, query); err != nil {
+			lastStmtError = err
+		}
+	}
+
+	// can't drop if not all privileges are revoked
+	if rows.Err() != nil {
+		return errwrap.Wrapf("could not generate revocation statements for all rows: {{err}}", rows.Err())
+	}
+	if lastStmtError != nil {
+		return errwrap.Wrapf("could not perform all revocation statements: {{err}}", lastStmtError)
+	}
+
+	// Drop this user
+	stmt, err = db.PrepareContext(ctx, fmt.Sprintf(
+		`DROP USER IF EXISTS %s;`, pq.QuoteIdentifier(username)))
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+	if _, err := stmt.ExecContext(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p *RedShift) RotateRootCredentials(ctx context.Context, statements []string) (map[string]interface{}, error) {
+	p.Lock()
+	defer p.Unlock()
+
+	if len(p.Username) == 0 || len(p.Password) == 0 {
+		return nil, errors.New("username and password are required to rotate")
+	}
+
+	rotateStatents := statements
```

Comment:
Looks like the var is misspelled, `s/rotateStatents/rotateStatements/`.

Reply:
A transient artifact no less https://github.com/hashicorp/vault/blob/master/plugins/database/postgresql/postgresql.go#L452

I'll fix this and PR a fix in the original codebase too. Good catch, thanks!

Reply:
https://github.com/hashicorp/vault/pull/8306